backend/server/public/src/index.jsx

```diff
@@ -3,12 +3,23 @@ import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
 import thunkMiddleware from 'redux-thunk';
 import { Provider, connect } from 'preact-redux';
 import { createLogger } from 'redux-logger';
-import { NEW_ADS, refresh, search } from 'utils.js';
+import { IS_LAST_PAGE, NOT_LAST_PAGE, PAGE_PREV, PAGE_NEXT, PAGE_CLEAR, NEW_ADS, pageCount, refresh, search } from 'utils.js';
```

Comment:
This is a big import. It would be better to throw `lastPage` and `pageIndex` into a pagination.js file and import from there, there's no real need for these action types to be in this scope.